### PR TITLE
[YAML] Implement a consistent BigQuery read and write transform.

### PR DESCRIPTION
**Pull request by robertwb**
This adapts the Java and Python BigQuery IOs to have the same interface and be callable from Yaml.

Several bugfixes were required along the way to make this work smoothly which are divided out in the commit history. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @Polber

File: sdks/python/apache_beam/yaml/standard_io.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      'WriteToBigQuery':
+        table: 'table'
+        create_disposition: 'createDisposition'
+        write_disposition: 'writeDisposition'
+        error_handling: 'errorHandling'
+    underlying_provider:
+      type: beamJar
+      transforms:
+        'ReadFromBigQuery': 'beam:schematransform:org.apache.beam:bigquery_storage_read:v1'
+        'WriteToBigQuery': 'beam:schematransform:org.apache.beam:bigquery_storage_write:v1'
```
Comment (Polber) [inline sdks/python/apache_beam/yaml/standard_io.yaml:45]: Should this be `v2`?

**Review by Polber — APPROVED** [review]

File: sdks/python/apache_beam/yaml/standard_io.yaml (hunk 1 above)
Comment (robertwb) [reply sdks/python/apache_beam/yaml/standard_io.yaml:45]: Oh, yes. (I changed this at the last minute just to be safe...)

**Comment by robertwb** [comment]
Java_Examples_Dataflow_Java17 PreCommit

**Comment by robertwb** [comment]
Run Python_Integration PreCommit

**Comment by robertwb** [comment]
Python failures are timeout flakes seen elsewhere, Java failures also unrelated (only java change is BQ write provider, which is unused in WordCount).

**Pull request merged**